xcm/xcm-builder/src/universal_exports.rs

```diff
@@ -352,8 +357,16 @@ impl<Router: SendXcm, OurPlace: Get<InteriorMultiLocation>> DispatchBlob
 			.map_err(|()| DispatchBlobError::NonUniversalDestination)?;
 		ensure!(intended_global == our_global, DispatchBlobError::WrongGlobal);
 		let dest = universal_dest.relative_to(&our_universal);
-		let message: Xcm<()> =
+		let mut message: Xcm<()> =
 			message.try_into().map_err(|_| DispatchBlobError::UnsupportedXcmVersion)?;
+
+		// Prepend our bridge instance discriminator.
+		// Can be used for fine-grained control of origin on destination in case of multiple bridge instances,
+		// e.g. restrict `type UniversalAliases` and `UniversalOrigin` instruction to trust just particular bridge instance for `NetworkId`.
+		if let Some(bridge_instance) = OurPlaceBridgeInstance::get() {
+			message.0.insert(0, DescendOrigin(bridge_instance));
```

Comment:
Is the 0th index the right location for `DescendOrigin`? Or better yet, is it the right instruction to add? IIRC the first XCM instruction is `UniversalOrigin`, so maybe we should check for that and append the bridge instance junction to it, or insert the `DescendOrigin` instruction right after the `UniversalOrigin` instruction.

Reply:
yes, we need to add `DescendOrigin(bridge_instance)` before any `UniversalOrigin`,

`BridgeBlobDispatcher` sends bridged xcm message to some `Router`,
e.g. in our case, BridgeHub sends this message over XCMP queue to AssetHub,
and on AssetHub, when `XcmExecutor` starts to execute xcm, the origin is initially `BridgeHubMultiLocation`,
so with this `DescendOrigin(bridge_instance)` we change origin to `BridgeHubMultiLocation+bridge_instance`,
so then we execute `UniversalOrigin(network)`, https://github.com/paritytech/polkadot/blob/master/xcm/xcm-executor/src/lib.rs#L813-L818  and we want to check here `UniversalAliases` for `(BridgeHubMultiLocation+bridge_instance, network)`